client/galaxy/scripts/mvc/ui/ui-misc.js

```diff
@@ -78,6 +78,57 @@ define(['utils/utils',
         }
     });
 
+
+    	var Password = Backbone.View.extend({
```

Comment:
In your example, Ui.Input is returned from _fieldText and in Bjoern's implementation I don't see the type changed to password in ui-misc.js, unless I am missing something. Is this what you are referring to?